ocaml/xapi/xapi_cluster.ml

```diff
@@ -71,7 +71,11 @@ let create ~__context ~network ~cluster_stack ~pool_auto_join ~token_timeout ~to
         set_ha_cluster_stack ~__context;
         cluster_ref
       | Result.Error error ->
-        D.warn "Error occurred during Cluster.create";
+        D.warn "Error occurred during Cluster.create. Cleaning up...";
+        Db.Cluster_host.destroy ~__context ~self:cluster_host_ref;
```

Comment:
Does this call fail if the ref isn't in the DB yet? In that case we should ignore exceptions that are raised.